middleware/middleware-1.3.16.md

```diff
@@ -0,0 +1,72 @@
+|----------------------------------------------|-------------------------------------|
+| fiskaltrust.Middleware.SCU.DE.Fiskaly        | BSI-K-TR-0403 [TSE in Evaluierung]  |
+| fiskaltrust.Middleware.SCU.DE.SwissbitCloud  | BSI-K-TR-0456-2021 [USK ausgesetzt] |
+| fiskaltrust.Middleware.SCU.DE.DeutscheFiskal | BSI-K-TR-0457-2021 [USK ausgesetzt] |
+
+## Stability improvement: Fiskal Cloud Connector
+We've updated our hosting of the _Fiskal Cloud Connector_ (FCC). This background service that is required by _Deutsche Fiskal_ and _Swissbit Cloud_ TSEs is automatically downloaded, initialized and started by our Middleware. However, we did not handle some initialization errors, leading to confusing errors for our users. We've changed the behavior so that the SCU now throws an exception and stops at startup time in case the FCC cannot initialized properly. If this happens, the error logs of the FCC are now printed into the console/our log file, allowing easier error diagnostics.
+
+In addition, we've changed the default directory for the FCC installation from `%appdata%/fiskaltrust/FCC` to our regular service folder. As the Middleware now uses the package installation for both Deutsche Fiskal and Swissbit Cloud (introduced in [1.3.14](middleware-1.3.14.md)), binding instances to users is not required anymore. This  should make testing and installing the FCC with different/default service users easier. **Existing installations are not affected by this change and continue to work.**
+
+## Compatibility improvement: Fiskal Cloud Connector on Linux
+During the last two weeks, our partners informed us that the FCC could not be successfully installed on some Linux distributions without manually making some files executable. We have changed this behavior so that no manual action is needed anymore when setting up a _Swissbit Cloud_ or a _Deutsche Fiskal_ TSE on Linux.
+
+## Bug fix: Deleting logs on fiskaly TSEs
+To ensure equal functionality in all our TSE implementations, we've updated our fiskaly SCU to now mark logs that have been exported during a daily closing receipt as deleted. This resolves two issues:
+- Previously, the fiskaly SCU always did a full export when daily-closing receipts were processed, and stored the results in the database. When a range-based `/Journal` export of those TAR files was called, the response could hence include more logs than requested. 
+- Due to the full export, the duration of daily-closing receipts on fiskaly SCUs increased with each processed transaction, ultimately making the process very slow when many transactions were performed. 
+
+Summarized, the fiskaly SCU now behaves as our other TSE implementations and only exports TAR logs that were generated since the last daily-closing receipt. However, there's one exception: in case there are open transactions, it's not possible to mark any transactions as deleted; the open transactions need to be closed first.
```

Comment:
```suggestion
Summarized, the fiskaly SCU now behaves as our other TSE implementations and only exports TAR logs that were generated since the last daily-closing receipt. However, there's one exception: similar to the behavior of the hardware TSEs in the case of open transactions, it's not possible to mark any transactions as deleted; the open transactions need to be closed first.
```